tests/scenarios/test_withads_search.py

```diff
@@ -59,11 +67,20 @@ def search_using_search_bar_to_return_ads(selenium):
         search_bar.send_keys(Keys.ENTER)
 
 
-@then("The browser reports correct telemetry for the search event")
-def check_telemetry_for_with_ads_url_search(find_ads_search_telemetry):
-    find_ads_search_telemetry("browser.search.withads.urlbar", ping_data={"google:tagged": 1})
+@given("The user highlights some text and wants to search for it via the context menu")
+def search_using_context_click_menu(selenium, simplehttpserver):
+    selenium.get("http://localhost:8000")
+    el = selenium.find_element(By.CSS_SELECTOR, "#search-to-return-ads")
+
+    ActionChains(selenium).move_to_element(el).pause(1).double_click(el).pause(1).context_click(
+        el
+    ).perform()
+    with selenium.context(selenium.CONTEXT_CHROME):
+        menu = selenium.find_element(By.CSS_SELECTOR, "#contentAreaContextMenu")
+        menu.find_element(By.CSS_SELECTOR, "#context-searchselect").click()
+    WebDriverWait(selenium, 60).until(EC.number_of_windows_to_be(3))
 
 
-@then("The browser reports correct telemetry for the searchbar event")
-def check_telemetry_for_with_ads_search_bar(find_ads_search_telemetry):
-    find_ads_search_telemetry("browser.search.withads.searchbar", ping_data={"google:tagged": 1})
+@then(parsers.parse("The browser reports correct telemetry for the {search:l} search event"))
```

Comment:
Had to do some googling for this, but verrry nice! 👏 
https://pytest-bdd.readthedocs.io/en/stable/#step-arguments